dataikuapi/dss/codestudio.py

```diff
@@ -0,0 +1,141 @@
+        self.client = client
+        self.project_key = project_key
+        self.code_studio_object_id = code_studio_object_id
+
+    def delete(self):
+        """
+        Delete the code studio
+        """
+        self.client._perform_empty("DELETE", "/projects/%s/code-studios/%s" % (self.project_key, self.code_studio_object_id))
+
+    def get_settings(self):
+        """
+        Get the code studio object's settings
+
+        :returns: a handle to manage the code studio settings
+        :rtype: :class:`dataikuapi.dss.codestudio.DSSCodeStudioObjectSettings`
+        """
+        settings = self.client._perform_json("GET", "/projects/%s/code-studios/%s" % (self.project_key, self.code_studio_object_id))
+        return DSSCodeStudioObjectSettings(self.client, self.project_key, self.code_studio_object_id, settings)
+
+    def get_status(self):
+        """
+        Get the code studio object's state
+
+        :returns: a handle to inspect the code studio state
+        :rtype: :class:`dataikuapi.dss.codestudio.DSSCodeStudioObjectStatus`
+        """
+        status = self.client._perform_json("GET", "/projects/%s/code-studios/%s/status" % (self.project_key, self.code_studio_object_id))
+        return DSSCodeStudioObjectStatus(self.client, self.project_key, self.code_studio_object_id, status)
+
+    def stop(self):
+        """
+        Stop a running code studio
+
+        :returns: a future to wait on the stop, or None if already stopped
+        :rtype: :class:`dataikuapi.dss.future.DSSFuture`
+        """
+        ret = self.client._perform_json("POST", "/projects/%s/code-studios/%s/stop" % (self.project_key, self.code_studio_object_id))
+        if 'jobId' in ret:
+            return self.client.get_future(ret["jobId"])
+        else:
+            return None
+
+    def restart(self):
+        """
+        Restart a code studio
+
+        :returns: a future to wait on the start
+        :rtype: :class:`dataikuapi.dss.future.DSSFuture`
+        """
+        ret = self.client._perform_json("POST", "/projects/%s/code-studios/%s/restart" % (self.project_key, self.code_studio_object_id))
+        if 'jobId' in ret:
+            return self.client.get_future(ret["jobId"])
+        else:
+            return None
+
+class DSSCodeStudioObjectSettings(object):
+    """
+    Settings for the code studio object
+    """
+    def __init__(self, client, project_key, code_studio_object_id, settings):
+        """Do not call directly, use :meth:`dataikuapi.dss.codestudio.DSSCodeStudioObject.get_settings`"""
+        self.client = client
+        self.project_key = project_key
+        self.code_studio_object_id = code_studio_object_id
+        self.settings = settings
+
+    def get_raw(self):
+        """
+        Gets all settings as a raw dictionary. This returns a reference to the raw settings, not a copy,
+        """
+        return self.settings
```

Comment:
save ?

Reply:
nothing to actually change at the moment, this call merely returns the code studio definition, so you can see the templateId and the libName. There is a `params` with a `runAs` field that's not implemented / used at the moment, but maybe one day...
So today the object is more a "immutable definition" than a "settings", but is named settings for future-proof-ness